Tests/EdgeCLITests/AsyncHTTPClientWrapperIntegrationTests.swift

```diff
@@ -0,0 +1,412 @@
+                expectingStatus: .noContent
+            )
+            
+            #expect(response.status == .noContent)
+            #expect(data.count == 0)  // Should be empty
+            
+        } catch {
+            print("Skipping empty response test due to network unavailability: \(error)")
+        }
+    }
+    
+    @Test("executeRequestThrowing handles HEAD requests correctly")
+    func testHeadRequestHandling() async throws {
+        let wrapper = AsyncHTTPClientWrapper()
+        
+        let request = HTTPRequest(
+            method: .head,
+            url: URL(string: "https://httpbin.org/get")!
+        )
+        
+        do {
+            let (data, response) = try await wrapper.executeRequestThrowing(
+                request,
+                expectingStatus: .ok
+            )
+            
+            #expect(response.status == .ok)
+            #expect(data.count == 0)  // HEAD responses should have no body
+            
+        } catch {
+            print("Skipping HEAD request test due to network unavailability: \(error)")
+        }
+    }
+    
+    // MARK: - Container Registry Specific Tests
+    
+    @Test("executeRequestThrowing handles Docker registry content types")
+    func testDockerRegistryContentTypes() async throws {
+        let wrapper = AsyncHTTPClientWrapper()
+        
+        let request = HTTPRequest.get(
+            URL(string: "https://httpbin.org/response-headers")!,
+            accepting: [
+                "application/vnd.docker.distribution.manifest.v2+json",
+                "application/vnd.docker.distribution.manifest.v1+prettyjws"
+            ]
+        )
+        
+        do {
+            let (data, response) = try await wrapper.executeRequestThrowing(
+                request,
+                expectingStatus: .ok
+            )
+            
+            #expect(response.status == .ok)
+            #expect(data.count > 0)
+            
+        } catch {
+            print("Skipping Docker content type test due to network unavailability: \(error)")
+        }
+    }
+}
+
+// MARK: - Performance Tests
+
+@Suite("AsyncHTTPClientWrapper Performance Tests")
+struct AsyncHTTPClientWrapperPerformanceTests {
+    
+    @Test("wrapper has minimal overhead compared to raw AsyncHTTPClient", .disabled("Performance test"))
+    func testPerformanceOverhead() async throws {
+        // This would be a performance comparison test
+        // Disabled by default as it's not essential for correctness
+        
+        let wrapper = AsyncHTTPClientWrapper()
+        let request = HTTPRequest.get(URL(string: "https://httpbin.org/get")!)
+        
+        let startTime = Date()
+
+        do {
+            _ = try await wrapper.executeRequestThrowing(request, expectingStatus: .ok)
+
+            let elapsed = Date().timeIntervalSince(startTime)
+
+            // Verify reasonable performance (< 5 seconds for a simple request)
+            #expect(elapsed < 5.0)
```

Comment:
I wouldn't normally test performance this way, as network latency is very circumstantial.

Reply:
I'll leave this for now, doesn't cause any problems and we can revisit